flake.nix

```diff
@@ -0,0 +1,32 @@
+{
+  description = "Future-proof note-taking and publishing based on Zettelkasten";
+
+  inputs = {
+    # Please sync revision with ./dep/nixpkgs/github.json
+    nixpkgs.url =
+      "github:nixos/nixpkgs/fcab19deb78fbb5ea24e19b133cf34358176396a";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; };
+
+        project = import ./project.nix { inherit pkgs; };
```

Comment:
I believe this should inherit `system` as well.

Reply:
if i understand correctly, only `default.nix` accepts `system` attribute and its only purpose is to eval `import ./dep/nixpkgs { inherit system; }`, which is same as `import nixpkgs { inherit system; }` in flake.

Is that the case?

Reply:
Right, n/m then.